src/java/main/org/apache/zookeeper/server/command/FourLetterCommands.java

```diff
@@ -153,13 +155,33 @@
      */
     public final static int telnetCloseCmd = 0xfff4fffd;
 
-    final static HashMap<Integer, String> cmd2String =
-        new HashMap<Integer, String>();
+    private static final String ZOOKEEPER_4LW_COMMANDS_WHITELIST = "zookeeper.4lw.commands.whitelist";
+
+    final static Map<Integer, String> cmd2String = new HashMap<Integer, String>();
+
+    final static Set<String> whiteListedCommands = new HashSet<String>();
 
     public static Map<Integer, String> getCmdMapView() {
         return Collections.unmodifiableMap(cmd2String);
     }
 
+    // ZOOKEEPER-2693: Only allow white listed commands.
+    public static Set<String> getWhiteListedCmdView() {
```

Comment:
The class name already provide context on caller site - FourLetterCommands.getWhiteListedFourLetterCmd sounds redundant.